app/app/src/main/java/com/example/clubwat/repository/DiscussionRepository.kt

```diff
@@ -0,0 +1,112 @@
+package com.example.clubwat.repository
+
+import com.example.clubwat.BuildConfig
+import com.example.clubwat.model.ClubDetails
+import com.example.clubwat.model.DiscussionData
+import com.example.clubwat.model.NetworkResult
+import com.example.clubwat.model.SendDiscusionMessageRequest
+import com.google.gson.Gson
+import org.json.JSONObject
+import java.io.OutputStreamWriter
+import java.net.HttpURLConnection
+import java.net.URL
+
+interface DiscussionRepository {
+    suspend fun getClub(clubId: String, userId: String): NetworkResult<ClubDetails>
+
+    suspend fun getPosts(clubId: String, userId: String): NetworkResult<DiscussionData>
+
+    suspend fun sendMessage(request: SendDiscusionMessageRequest, bearer: String): NetworkResult<Any>
+}
+
+class DiscussionRepositoryImpl : DiscussionRepository {
+    override suspend fun getClub(clubId: String, userId: String): NetworkResult<ClubDetails> {
```

Comment:
There is a similar function in ClubDetailsView. Should we put the function in a different file or just keep it in both places?

Reply:
The previous same network call is in a viewmodel, which I can't reuse in this screen. So these network calls are now in a repository that can be reused by any viewmodel.